app/controllers/application_controller.rb

```diff
@@ -5,6 +5,137 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      redirect :'tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params[:username].empty? || params[:email].empty? || params[:password].empty?
+      redirect '/signup'
+    else
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect :'tweets'
+    end
+  end
+
+  get '/login' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      redirect :'tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    #binding.pry
+    @user = User.find_by(username: params[:username])
+    if !@user.nil?
+    #  @tweets = Tweet.all
+      session[:user_id] = @user.id
+
+      redirect :'tweets'
+    else
+      erb :'users/signup'
+    end
+  end
+
+  get '/tweets' do
+    #binding.pry
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect :'users/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      erb :'tweets/create_tweet'
+    else
+      redirect :'users/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      @tweet = Tweet.find(params[:id])
+      #binding.pry
+      erb :'tweets/show_tweet'
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if Helpers.is_logged_in?(session)
+      @user = Helpers.current_user(session)
+      @tweet = Tweet.find(params[:id])
+      #binding.pry
+      erb :'tweets/edit_tweet'
+    else
+      redirect :'users/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].empty?
+      redirect :'tweets/new'
+    else
+      @tweet = Tweet.create(content: params[:content], user_id: session[:user_id])
+    end
+  end
+
+  post '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+    if params[:content].empty?
+      redirect :"tweets/#{@tweet.id}/edit"
+    else
+      @tweet.update(content: params[:content])
+    end
+    #binding.pry
+  end
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find(params[:id])
+    #binding.pry
+    if @tweet.user == Helpers.current_user(session)
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.